core/src/main/java/com/netflix/conductor/core/execution/mapper/SubWorkflowTaskMapper.java

```diff
@@ -64,19 +57,17 @@ public SubWorkflowTaskMapper(ParametersUtils parametersUtils, MetadataDAO metada
 
         String subWorkflowName = resolvedParams.get("name").toString();
 
-        Integer subWorkflowVersion = getSubWorkflowVersion(resolvedParams, subWorkflowName);
-
         Task subWorkflowTask = new Task();
         subWorkflowTask.setTaskType(SubWorkflow.NAME);
         subWorkflowTask.setTaskDefName(taskToSchedule.getName());
         subWorkflowTask.setReferenceTaskName(taskToSchedule.getTaskReferenceName());
         subWorkflowTask.setWorkflowInstanceId(workflowInstance.getWorkflowId());
-        subWorkflowTask.setWorkflowType(workflowInstance.getWorkflowType());
+        subWorkflowTask.setWorkflowType(workflowInstance.getWorkflowName());
         subWorkflowTask.setCorrelationId(workflowInstance.getCorrelationId());
         subWorkflowTask.setScheduledTime(System.currentTimeMillis());
         subWorkflowTask.setEndTime(System.currentTimeMillis());
         subWorkflowTask.getInputData().put("subWorkflowName", subWorkflowName);
-        subWorkflowTask.getInputData().put("subWorkflowVersion", subWorkflowVersion);
+        subWorkflowTask.getInputData().put("subWorkflowVersion", subWorkflowParams.getVersion());
```

Comment:
is it safe to use this and remove existing `getSubWorkflowVersion` method?

Reply:
The subworkflowParamVersion is now populated at the beginning of the process (MetadataMapperService). Wrote tests for that in that class. I believe there are tests also on the legacy ones that handle subworkflow task creations.
Are you thinking on a specific scenario?